src/view.tree.js

```diff
@@ -52,28 +52,11 @@ class TreeView {
   _showHeader(repo) {
     const adapter = this.adapter
 
-    let currentBranchName = repo.branch.toString();
-    let branchHtml = '';
-
-    // Display branch switcher if the repository has multiple branches
-    if (repo.branches && repo.branches.length > 1) {
-      branchHtml = '<select id="octotree_header_branch_switcher">' +
-        repo.branches.map((branchName) => {
-          const safeBranchName = this._deXss(branchName).toString();
-
-          // Disable non-matching branch names, because URL won't work
-          const disabled = branchName !== safeBranchName;
-          const selected = branchName === currentBranchName;
-
-          return '<option value="' + safeBranchName + '"' +
-            (disabled ? ' disabled' : '') +
-            (selected ? ' selected' : '') +
-            '>' + safeBranchName + '</option>'
-        }).join('') +
-      '</select>';
-    }
-    else {
-      branchHtml = this._deXss(currentBranchName);
+    // Copy GitHub's branch switcher
+    let branchHtml = $('<div>').append($('.branch-select-menu').clone()).html();
```

Comment:
What is the page in question doesn't have the selector? Should we check if the element exist first and use fallback (i.e. no selector) if not? Also, should the element be styled/customized a bit so that it looks more fit in the octotree header?